src/python/pants/rules/core/fmt.py

```diff
@@ -69,25 +76,30 @@ class Fmt(Goal):
   workspace: Workspace,
   union_membership: UnionMembership
 ) -> Fmt:
-  nested_results = await MultiGet(
-    Get[FmtResults](FormatTarget, target.adaptor)
+  aggregated_results = await MultiGet(
+    Get[AggregatedFmtResults](FormatTarget, target.adaptor)
     for target in targets
     if FormatTarget.is_formattable(target.adaptor, union_membership=union_membership)
   )
-  results = [result for results in nested_results for result in results]
+  individual_results = [
+    result
+    for aggregated_result in aggregated_results
+    for result in aggregated_result.results
+  ]
 
-  if not results:
+  if not individual_results:
     return Fmt(exit_code=0)
 
   # NB: this will fail if there are any conflicting changes, which we want to happen rather than
-  # silently having one result override the other.
-  # TODO(#8722): get this working with multiple formatters for the same language. Right now, the
-  #  rule will fail if formatters touch the same file.
+  # silently having one result override the other. In practicality, this should never happen due
```

Comment:
> In practicality, this should never happen

I'm not sure I understand this comment. isort and black can and will attempt to modify the same files, for example. To mitigate this in some testing I was doing last week I explicitly removed the check for conflicting files when merging snapshots in a dev branch, which was necessary because this conflict would happen. Is there a reason we don't expect this to occur?

Reply:
> isort and black can and will attempt to modify the same files, for example.

Yes, this is why we use an aggregator for each language. Python has an aggregator `python_fmt_target.py` that will run all the relevant formatters over Python targets _sequentially_ and merge results so that there is no conflict.

So long as the aggregators are implemented correctly, i.e. each language is distinct and does not touch the other, like the Shell formatters don't touch Python files and vice versa, then there will "in practicality" be no merge conflict. If there is a merge conflict, then we want the engine to fail so that we are aware of this situation.

Reply:
Ok, I misunderstood the issue! Thank you!

Reply:
No worries! It took me a really long time to figure this out (thanks to your, Stu’s, and Daniel’s help!!). 

I believe this is a signal that the comment might not be as clear as I hoped. Normally I’d try improving it but am trying to land this in the RC tonight and mostly afk now. Hope that what’s up is okay.

Reply:
I think the PR description has enough information!